File: src/firefly/js/visualize/Band.js
/*
 * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
 */
import Enum from 'enum';

/**
 * @typedef {Object} Band
 * A Band in an image plot
 * must be enum 'RED', 'GREEN', 'BLUE', 'NO_BAND'
 * @prop RED
 * @prop GREEN
 * @prop BLUE
 * @prop NO_BAND
 * @type {Enum}
 * @public
 * @global
 */

/** @type {Band} */
export const Band= new Enum({'RED':0, 'GREEN':1, 'BLUE':2, 'NO_BAND':0}, {ignoreCase:true});

/** @type {Array.<Band>} */
export const allBandAry= [Band.RED,Band.GREEN,Band.BLUE];

File: src/firefly/java/edu/caltech/ipac/firefly/visualize/Band.java
/*
 * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
 */
package edu.caltech.ipac.firefly.visualize;

public enum Band {
    RED("Red", 0),
    GREEN("Green", 1),
    BLUE("Blue", 2),
    NO_BAND("NO_BAND", 0);
    private final String desc;
    private final int idx;

    Band(String desc, int idx) {
        this.desc = desc;
        this.idx = idx;
    }

    public String toString() { return desc; }
    public int getIdx() { return idx; }

    public static Band parse(String s) {
        try {
            return Enum.valueOf(Band.class,s.toUpperCase());
        } catch (Exception ignore) {
            return null;
        }
    }
}